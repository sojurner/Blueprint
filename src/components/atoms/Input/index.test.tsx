import React from 'react';
import { mount } from 'enzyme';
import LystInput from '.';

const wrap = (
  props = {
    inputProps: {
      styles: ''
    }
  }
) => mount(<LystInput {...props} />);

describe('LystInput', () => {
  let wrapper = wrap();

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot when styles are given', () => {
    const props = {
      inputProps: {
        styles: 'margin 1rem'
      }
    };
    wrapper = wrap(props);
    expect(wrapper).toMatchSnapshot();
  });
});
