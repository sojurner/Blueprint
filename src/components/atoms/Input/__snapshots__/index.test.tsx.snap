// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LystInput should match snapshot 1`] = `
<LystInput
  inputProps={
    Object {
      "styles": "",
    }
  }
>
  <WithStyles(ForwardRef(TextField))
    styles=""
  >
    <ForwardRef(TextField)
      classes={
        Object {
          "root": "MuiTextField-root",
        }
      }
      styles=""
    >
      <WithStyles(ForwardRef(FormControl))
        className="MuiTextField-root"
        required={false}
        styles=""
        variant="standard"
      >
        <ForwardRef(FormControl)
          className="MuiTextField-root"
          classes={
            Object {
              "fullWidth": "MuiFormControl-fullWidth",
              "marginDense": "MuiFormControl-marginDense",
              "marginNormal": "MuiFormControl-marginNormal",
              "root": "MuiFormControl-root",
            }
          }
          required={false}
          styles=""
          variant="standard"
        >
          <div
            className="MuiFormControl-root MuiTextField-root"
            styles=""
          >
            <WithStyles(ForwardRef(Input))>
              <ForwardRef(Input)
                classes={
                  Object {
                    "disabled": "Mui-disabled",
                    "error": "Mui-error",
                    "focused": "Mui-focused",
                    "formControl": "MuiInput-formControl",
                    "fullWidth": "MuiInput-fullWidth",
                    "input": "MuiInput-input",
                    "inputMarginDense": "MuiInput-inputMarginDense",
                    "inputMultiline": "MuiInput-inputMultiline",
                    "inputTypeSearch": "MuiInput-inputTypeSearch",
                    "multiline": "MuiInput-multiline",
                    "root": "MuiInput-root",
                    "underline": "MuiInput-underline",
                  }
                }
              >
                <WithStyles(ForwardRef(InputBase))
                  classes={
                    Object {
                      "disabled": "Mui-disabled",
                      "error": "Mui-error",
                      "focused": "Mui-focused",
                      "formControl": "MuiInput-formControl",
                      "fullWidth": "MuiInput-fullWidth",
                      "input": "MuiInput-input",
                      "inputMarginDense": "MuiInput-inputMarginDense",
                      "inputMultiline": "MuiInput-inputMultiline",
                      "inputTypeSearch": "MuiInput-inputTypeSearch",
                      "multiline": "MuiInput-multiline",
                      "root": "MuiInput-root MuiInput-underline",
                      "underline": null,
                    }
                  }
                  fullWidth={false}
                  inputComponent="input"
                  multiline={false}
                  type="text"
                >
                  <ForwardRef(InputBase)
                    classes={
                      Object {
                        "adornedEnd": "MuiInputBase-adornedEnd",
                        "adornedStart": "MuiInputBase-adornedStart",
                        "disabled": "Mui-disabled Mui-disabled",
                        "error": "Mui-error Mui-error",
                        "focused": "Mui-focused Mui-focused",
                        "formControl": "MuiInputBase-formControl MuiInput-formControl",
                        "fullWidth": "MuiInputBase-fullWidth MuiInput-fullWidth",
                        "input": "MuiInputBase-input MuiInput-input",
                        "inputAdornedEnd": "MuiInputBase-inputAdornedEnd",
                        "inputAdornedStart": "MuiInputBase-inputAdornedStart",
                        "inputMarginDense": "MuiInputBase-inputMarginDense MuiInput-inputMarginDense",
                        "inputMultiline": "MuiInputBase-inputMultiline MuiInput-inputMultiline",
                        "inputTypeSearch": "MuiInputBase-inputTypeSearch MuiInput-inputTypeSearch",
                        "marginDense": "MuiInputBase-marginDense",
                        "multiline": "MuiInputBase-multiline MuiInput-multiline",
                        "root": "MuiInputBase-root MuiInput-root MuiInput-underline",
                      }
                    }
                    fullWidth={false}
                    inputComponent="input"
                    multiline={false}
                    type="text"
                  >
                    <div
                      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                      onClick={[Function]}
                    >
                      <input
                        aria-invalid={false}
                        className="MuiInputBase-input MuiInput-input"
                        disabled={false}
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        required={false}
                        type="text"
                      />
                    </div>
                  </ForwardRef(InputBase)>
                </WithStyles(ForwardRef(InputBase))>
              </ForwardRef(Input)>
            </WithStyles(ForwardRef(Input))>
          </div>
        </ForwardRef(FormControl)>
      </WithStyles(ForwardRef(FormControl))>
    </ForwardRef(TextField)>
  </WithStyles(ForwardRef(TextField))>
</LystInput>
`;

exports[`LystInput should match snapshot when styles are given 1`] = `
<LystInput
  inputProps={
    Object {
      "styles": "margin 1rem",
    }
  }
>
  <Styled(WithStyles(ForwardRef(TextField)))
    styles="margin 1rem"
  >
    <WithStyles(ForwardRef(TextField))
      className="sc-bdVaJa dFnbXv"
      styles="margin 1rem"
    >
      <ForwardRef(TextField)
        className="sc-bdVaJa dFnbXv"
        classes={
          Object {
            "root": "MuiTextField-root",
          }
        }
        styles="margin 1rem"
      >
        <WithStyles(ForwardRef(FormControl))
          className="MuiTextField-root sc-bdVaJa dFnbXv"
          required={false}
          styles="margin 1rem"
          variant="standard"
        >
          <ForwardRef(FormControl)
            className="MuiTextField-root sc-bdVaJa dFnbXv"
            classes={
              Object {
                "fullWidth": "MuiFormControl-fullWidth",
                "marginDense": "MuiFormControl-marginDense",
                "marginNormal": "MuiFormControl-marginNormal",
                "root": "MuiFormControl-root",
              }
            }
            required={false}
            styles="margin 1rem"
            variant="standard"
          >
            <div
              className="MuiFormControl-root MuiTextField-root sc-bdVaJa dFnbXv"
              styles="margin 1rem"
            >
              <WithStyles(ForwardRef(Input))>
                <ForwardRef(Input)
                  classes={
                    Object {
                      "disabled": "Mui-disabled",
                      "error": "Mui-error",
                      "focused": "Mui-focused",
                      "formControl": "MuiInput-formControl",
                      "fullWidth": "MuiInput-fullWidth",
                      "input": "MuiInput-input",
                      "inputMarginDense": "MuiInput-inputMarginDense",
                      "inputMultiline": "MuiInput-inputMultiline",
                      "inputTypeSearch": "MuiInput-inputTypeSearch",
                      "multiline": "MuiInput-multiline",
                      "root": "MuiInput-root",
                      "underline": "MuiInput-underline",
                    }
                  }
                >
                  <WithStyles(ForwardRef(InputBase))
                    classes={
                      Object {
                        "disabled": "Mui-disabled",
                        "error": "Mui-error",
                        "focused": "Mui-focused",
                        "formControl": "MuiInput-formControl",
                        "fullWidth": "MuiInput-fullWidth",
                        "input": "MuiInput-input",
                        "inputMarginDense": "MuiInput-inputMarginDense",
                        "inputMultiline": "MuiInput-inputMultiline",
                        "inputTypeSearch": "MuiInput-inputTypeSearch",
                        "multiline": "MuiInput-multiline",
                        "root": "MuiInput-root MuiInput-underline",
                        "underline": null,
                      }
                    }
                    fullWidth={false}
                    inputComponent="input"
                    multiline={false}
                    type="text"
                  >
                    <ForwardRef(InputBase)
                      classes={
                        Object {
                          "adornedEnd": "MuiInputBase-adornedEnd",
                          "adornedStart": "MuiInputBase-adornedStart",
                          "disabled": "Mui-disabled Mui-disabled",
                          "error": "Mui-error Mui-error",
                          "focused": "Mui-focused Mui-focused",
                          "formControl": "MuiInputBase-formControl MuiInput-formControl",
                          "fullWidth": "MuiInputBase-fullWidth MuiInput-fullWidth",
                          "input": "MuiInputBase-input MuiInput-input",
                          "inputAdornedEnd": "MuiInputBase-inputAdornedEnd",
                          "inputAdornedStart": "MuiInputBase-inputAdornedStart",
                          "inputMarginDense": "MuiInputBase-inputMarginDense MuiInput-inputMarginDense",
                          "inputMultiline": "MuiInputBase-inputMultiline MuiInput-inputMultiline",
                          "inputTypeSearch": "MuiInputBase-inputTypeSearch MuiInput-inputTypeSearch",
                          "marginDense": "MuiInputBase-marginDense",
                          "multiline": "MuiInputBase-multiline MuiInput-multiline",
                          "root": "MuiInputBase-root MuiInput-root MuiInput-underline",
                        }
                      }
                      fullWidth={false}
                      inputComponent="input"
                      multiline={false}
                      type="text"
                    >
                      <div
                        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                        onClick={[Function]}
                      >
                        <input
                          aria-invalid={false}
                          className="MuiInputBase-input MuiInput-input"
                          disabled={false}
                          onBlur={[Function]}
                          onChange={[Function]}
                          onFocus={[Function]}
                          required={false}
                          type="text"
                        />
                      </div>
                    </ForwardRef(InputBase)>
                  </WithStyles(ForwardRef(InputBase))>
                </ForwardRef(Input)>
              </WithStyles(ForwardRef(Input))>
            </div>
          </ForwardRef(FormControl)>
        </WithStyles(ForwardRef(FormControl))>
      </ForwardRef(TextField)>
    </WithStyles(ForwardRef(TextField))>
  </Styled(WithStyles(ForwardRef(TextField)))>
</LystInput>
`;
